internal/xsql/collection.go

```diff
@@ -157,6 +158,22 @@
 	return nil
 }
 
+func (w *WindowTuples) RangeProjectSet(f func(i int, r CloneAbleRow) ([]CloneAbleRow, error)) ([]Collection, error) {
+	newCollections := make([]Collection, 0)
+	for i, oldTupleRow := range w.Content {
+		newTuples, err := f(i, oldTupleRow.Clone())
+		if err != nil {
+			return nil, err
+		}
+		for _, newTuple := range newTuples {
+			ns := w.Clone().(*WindowTuples)
```

Comment:
why return multiple collections? we can just flatten to a collection. It is also save to replace the old content